Add ELASTIC_APM_SERVICE_NODE_NAME configuration option (#616)
elastic/apm-agent-dotnet: docs/configuration.asciidoc

| Name of the entry assembly       | String 
|============

[float]
[[config-service-node-name]]

> ==== `ServiceNodeName` (added[1.2])

Reviewer: I personally linked the [Go version of this doc](https://github.com/elastic/apm-agent-go/pull/651/files#diff-01c1a79a5a1e1b67003a1885365f2cf5R153) better, but this is also totally fine.

> ==== `ServiceNodeName` (added[1.2])

Author: @gregkalapos, I believe we need to change version here, as this feature isn't included in 1.2 version. Maybe, we need to think about some placeholders which will be replaced during release to avoid issues with documentation

Reviewer: Thanks for tagging me @vhatsura! Opened https://github.com/elastic/apm-agent-dotnet/pull/627 - yeah we should came up with something for this.

> If set, this name is used to distinguish between different nodes of a service.

Reviewer 2: ```suggestion
Optional name used to differentiate between nodes in a service.
```

If not set, data aggregations will be done based on a container ID (where valid) or on the reported hostname (automatically discovered).

> NOTE: Metrics views can utilize this configuration since APM Server 7.5

Reviewer 2: ```suggestion
NOTE: This feature requires APM Server versions >= 7.5
```

[options="header"]
|============
| Environment variable name       | IConfiguration or Web.config key 

> | `ELASTIC_APM_SERVICE_NODE_NAME` | None

Reviewer: The `IConfiguration or Web.config key ` part is TBD, if we all agree we want to have that, this needs to be updated. If not, then sorry for the noise :)

Author: done
